### PR TITLE
tools/profile: add support for PID-namespacing

### PR DESCRIPTION
**Pull request by lucab**
This adds translation logic to perform PID remapping across PID-namespaces.
It is now possible to profile a target process from within a nested PID-namespace (e.g. in a container).

### CODE REVIEW

File: tools/profile.py [hunk 1]
```diff
@@ -216,6 +225,18 @@ def stack_id_err(stack_id):
 }
 """
 
+# pid-namespace translation
+try:
+    devinfo = os.stat("/proc/self/ns/pid")
+    bpf_text = bpf_text.replace('USE_PIDNS', "1")
+    bpf_text = bpf_text.replace('PIDNS_DEV', str(devinfo.st_dev))
+    bpf_text = bpf_text.replace('PIDNS_INO', str(devinfo.st_ino))
+    pidns_translation = "1"
```
Comment (chenhengqi) [inline tools/profile.py:234]: `pidns_translation` is unused.

Comment (lucab) [reply tools/profile.py]: Ah right sorry, it's leftover from a previous iteration of the patch, I dropped it now.

**Comment by yonghong-song** [comment]
Can we add an option for this new functionality, something like '--with-pid-namespace'? Note that with '--with-pid-namespace', the command line arguments like pid/tid should be the one in the pid namespace to make the tool work.
With introducing new option, please add a few examples in the example file and update the man page as well.

**Comment by yonghong-song** [comment]
Okay, @lucab Your argument above indeed makes sense. Without this symbolization will be incorrect in child pid namespace.

**Comment by lucab** [comment]
@yonghong-song good to hear, and thanks for taking the time to review this!

**Pull request merged**